DependencyUpdated/Program.cs

```diff
@@ -46,24 +47,57 @@ private static void RunApplication(Options options)
 
             foreach (var configEntry in config.Value.Projects)
             {
+                repositoryProvider.SwitchToDefaultBranch(repositoryPath);
+
                 var updater = _serviceProvider.GetRequiredKeyedService<IProjectUpdater>(configEntry.Type);
+               
                 foreach (var project in configEntry.Directories)
                 {
-                    repositoryProvider.SwitchToDefaultBranch(repositoryPath);
-                    var projectName = configEntry.Name;
-                    repositoryProvider.SwitchToUpdateBranch(repositoryPath, projectName);
-                  
-                    var updates = updater.UpdateProject(project, configEntry).Result;
+                    if (!Path.Exists(project))
+                    {
+                        throw new FileNotFoundException("Search path not found", project);
+                    }
+                    var allDepencenciesToUpdate = new List<DependencyDetails>();
+                    var projectFiles = updater.GetAllProjectFiles(project).ToArray(); 
+                   
+                    foreach (var projectFile in projectFiles)
+                    {
+                        var dependencyToUpdate = await updater.ExtractAllPackagesThatNeedToBeUpdated(projectFile, configEntry);
+                        allDepencenciesToUpdate.AddRange(dependencyToUpdate);
+                        
+                    }
 
-                    if (updates.Count == 0)
+                    if (allDepencenciesToUpdate.Count == 0) //no libs to update, we can skip this project
```

Comment:
Comment to delete, If you want to explain If statment -> extract that to method and name it - but IMO it's pretty cleare here.